src/VisualStudio/Core/Def/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -79,20 +84,17 @@ internal sealed class TaskCenterSolutionAnalysisProgressReporter
 
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public TaskCenterSolutionAnalysisProgressReporter(
-            SVsTaskStatusCenterService taskStatusCenterService,
-            IDiagnosticService diagnosticService,
-            VisualStudioWorkspace workspace)
+        public TaskCenterSolutionAnalysisProgressReporter(VisualStudioWorkspace workspace)
         {
-            _taskCenterService = (IVsTaskStatusCenterService)taskStatusCenterService;
-            _options = new TaskHandlerOptions()
-            {
-                Title = ServicesVSResources.Running_low_priority_background_processes,
-                ActionsAfterCompletion = CompletionActions.None
-            };
+            _workspace = workspace;
+        }
 
-            var crawlerService = workspace.Services.GetRequiredService<ISolutionCrawlerService>();
-            var reporter = crawlerService.GetProgressReporter(workspace);
+        public async Task InitializeAsync(IAsyncServiceProvider serviceProvider)
+        {
+            _taskCenterService = await serviceProvider.GetServiceAsync<SVsTaskStatusCenterService, IVsTaskStatusCenterService>().ConfigureAwait(false);
```

Comment:
📝 Make sure this line is _not_ using the `GetServiceAsync` extension method provided by the VS SDK.

Reply:
@AArnott ?

Reply:
It's true that it relies on `ThreadHelper`. If you're not using the vssdktestfx or something similar to make ThreadHelper work, then ya, this will probably render you code unable to run in your unit tests.